vignettes/parallel_computing_with_extensions.Rmd

```diff
@@ -0,0 +1,366 @@
+---
+title: "How to use parallel computing with extensions to the package, an example"
+author: "Oliver Boix"
+date: "14 March 2023"
+output: rmarkdown::html_vignette
+bibliography: vignettes.bib  
+vignette: >
+  %\VignetteIndexEntry{Parallel computing with extensions}
+  %\VignetteEngine{knitr::rmarkdown}
+  %\VignetteDepends{crmPack}
+  \usepackage[utf8]{inputenc}
+---
+
+```{r setup, include=FALSE}
+knitr::opts_chunk$set(
+  collapse = TRUE,
+  comment = "#>"
+)
+```
+
+## Introduction
+
+One of the big advantages of `crmPack` over existing R implementations is its flexible framework based on the S4 classes and methods system [@Chambers2008]. Users can extend the existing functionality easily to the specific needs of the study [@Bove2019].
+
+User defined extensions of classes and methods can be easily created and used together with existing `crmPack` classes and methods when setting up the study and performing analysis with observed data. As long as no parallel computing is used, no special care needs to be taken utilizing both native and user defined classes and methods.
+When trial simulations need to be performed to derive the operational characteristics of a study setup, run times may become long with single core processing. The run time depends mainly on the number of study replications and MCMC samples. In cases of long run times, utilizing parallel computing, i.e. using multiple CPU cores, can overcome these challenge and decrease run times significantly.
+
+Parallel computing is supported by `crmPack` by design, i.e. whenever the simulation call is invoked with parameters `parallel = TRUE` and `nCores =` more than one CPU, each core is initialized with the `crmPack` package and the global environment.
+
+Due to the nature of the S4 object system, user defined classes and method that were defined in the global environment cannot just made available at each core by initializing the core with the global environment. User defined classes and method needs to be executed on any core that is initialized to warrant full functionality of the S4 objects. 
+
+To use user written extension with parallel computing and `crmPack`, the user written code needs to be embedded in a function that must have the fixed name `crmpack_extensions`. This function is executed at any core at the time of initialization of the core, so that the user written S4 classes and methods are available along with the `crmPack` package.
```

Comment:
then you could adapt that here

Reply:
I revised the whole vignette accordingly.